animal-options.lua

```diff
@@ -0,0 +1,175 @@
+    else
+        unit.flags2.slaughter = false
+    end
+end
+
+-- set units marked for gelding flag
+function creature_screen:set_geld_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option == YES then
+        unit.flags3.marked_for_gelding = true
+    else
+        unit.flags3.marked_for_gelding = false
+    end
+end
+
+-- set available for adoption flag
+function creature_screen:set_adoption_flag(option)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    if not unit then return end
+
+    if option == YES then
+        unit.flags3.available_for_adoption = true
+    else
+        unit.flags3.available_for_adoption = false
+    end
+end
+
+-- Check current flag status of animal to dynamically set on/off
+function creature_screen:get_butcher(unit)
+    if not unit then return end
+
+    if unit.flags2.slaughter == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+function creature_screen:get_geld(unit)
+    if not unit then return end
+
+    if unit.flags3.marked_for_gelding == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+function creature_screen:get_adopt(unit)
+    if not unit then return end
+
+    if unit.flags3.available_for_adoption == true then
+        return YES
+    else
+        return NO
+    end
+end
+
+-- Use render to set On/Off dynamically for each unit
+function creature_screen:render(dc)
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    self.subviews.butcher_animal:setOption(self:get_butcher(unit))
+    self.subviews.geld_animal:setOption(self:get_geld(unit))
+    self.subviews.adopt_animal:setOption(self:get_adopt(unit))
+
+    creature_screen.super.render(self, dc)
+end
+
+function creature_screen:init()
+self:addviews{
+    widgets.Panel{
+        frame_background=gui.CLEAR_PEN,
+        frame_style=gui.FRAME_MEDIUM,
+        visible=check_valid_unit,
+        subviews={
+            widgets.CycleHotkeyLabel{
+                frame={t=0,l=0},
+                label='Butcher',
+                key='CUSTOM_CTRL_B',
+                options={
+                    {label=NO, value=NO, pen=COLOR_WHITE},
+                    {label=YES, value=YES, pen=COLOR_GREEN},
+                },
+                view_id='butcher_animal',
+                on_change=function(val) self:set_slaughter_flag(val) end,
+            },
+            widgets.ToggleHotkeyLabel{
+                frame={t=1,l=0},
+                label='Geld',
+                key='CUSTOM_CTRL_G',
+                options={
+                    {label=NO, value=NO, pen=COLOR_WHITE},
+                    {label=YES, value=YES, pen=COLOR_GREEN},
+                },
+                view_id='geld_animal',
+                enabled=is_geldable,
+                on_change = function(val) self:set_geld_flag(val) end
+            },
+            widgets.ToggleHotkeyLabel{
+                frame={t=2,l=0},
+                label='Adopt',
```

Comment:
also needs an `enabled` attribute that responses to `dfhack.units.isAvailableForAdoption(unit)`

Reply:
When I add that check it disables the option and seems to only reenable if i go into the creatures screen and turn on adoption there

Reply:
what is your `enable` logic?

Reply:
```
local function is_avail_adoption()
    local unit = dfhack.gui.getSelectedUnit(true)

    return dfhack.units.isAvailableForAdoption(unit)
end
```

Reply:
ohohoh, it returns whether the flag is already set, not whether it *can* be set. Hrm.. what's the logic for whether it *can* be set? just your existing validity check and race != cat?

Reply:
Oh i did not know cats couldn't be adopted (noob) ill add that in and that will end up being the check

Reply:
indeed, cats adopt dwarves, not the other way around